examples/thermostat/thermostat-common/include/atomic-write-manager.h

```diff
@@ -0,0 +1,186 @@
+     * @param subjectDescriptor The subject descriptor to check against
+     * @param endpoint The endpoint to check atomic write state
+     * @return true if there is an open atomic write
+     * @return false if there is not an open atomic write
+     */
+    bool InWrite(const std::optional<AttributeId> attributeId, const Access::SubjectDescriptor & subjectDescriptor,
+                 EndpointId endpoint) override;
+
+    /**
+     * @brief Check if there is an open atomic write on an endpoint, associated with the source node for a given command invocation,
+     * and optionally associated with a specific attribute
+     *
+     * @param attributeId
+     * @param commandObj
+     * @param endpoint
+     * @return true
+     * @return false
+     */
+    bool InWrite(const std::optional<AttributeId> attributeId, CommandHandler * commandObj, EndpointId endpoint) override;
+
+    /**
+     * @brief Check if there is an open atomic write on an endpoint, associated with the source node for a given command invocation,
+     * and associated with the set of attribute IDs
+     *
+     * @param attributeIds The set of attribute IDs to check against
+     * @param commandObj The command being invoked
+     * @param endpoint The endpoint for the atomic write
+     * @return true if there is an open atomic write
+     * @return false if there is not an open atomic write
+     */
+    bool InWrite(const DataModel::DecodableList<AttributeId>::Iterator attributeIds, CommandHandler * commandObj,
+                 EndpointId endpoint) override;
+
+    /**
+     * @brief Attempt to start an atomic write
+     *
+     * @param commandObj The AtomicRequest command
+     * @param commandPath The path for the command
+     * @param commandData The payload of the command
+     * @return true if the atomic write was opened
+     * @return false if the atomic write was not opened
+     */
+    bool BeginWrite(chip::app::CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                    const Commands::AtomicRequest::DecodableType & commandData) override;
+
+    /**
+     * @brief Attempt to commit an atomic write; returns true if the server is able to commit or failed trying, false if it was
+     * unable to find a matching atomic write
+     *
+     * @param commandObj The AtomicRequest command
+     * @param commandPath The path for the command
+     * @param commandData The payload of the command
+     * @return true if the atomic write was committed or rolled back
+     * @return false if the atomic write was not found
+     */
+    bool CommitWrite(chip::app::CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                     const Commands::AtomicRequest::DecodableType & commandData) override;
+
+    /**
+     * @brief Attempt to roll back an atomic write; returns true if the server is able to commit or failed trying, false if it was
+     * unable to find a matching atomic write
```

Comment:
same comment as above. maybe better to say returns false otherwise (unless there's no other scenarios that could occur). applies to other instances in this file.

Reply:
Rewrote to have the bool just mean whether or not the command was handled, which is more in line with how ember callbacks work.